src/Language/PureScript/Make/BuildPlan.hs

```diff
@@ -117,37 +121,33 @@ construct MakeActions{..} (sorted, graph) = do
       buildJob <- BuildJob <$> C.newEmptyMVar <*> C.newEmptyMVar
       pure (M.insert moduleName buildJob prev)
 
-    findExistingExtern :: M.Map ModuleName Prebuilt -> Module -> m (M.Map ModuleName Prebuilt)
-    findExistingExtern prev (getModuleName -> moduleName) = do
-      outputTimestamp <- getOutputTimestamp moduleName
+    findExistingExtern :: Module -> m (Maybe (ModuleName, Prebuilt))
+    findExistingExtern (getModuleName -> moduleName) = runMaybeT $ do
+      outputTimestamp <- MaybeT $ getOutputTimestamp moduleName
+      inputTimestamp <- lift $ getInputTimestamp moduleName
+      existingTimestamp <- MaybeT $ pure $ case inputTimestamp of
+        Left RebuildNever ->
```

Comment:
Regarding the test failures, perhaps the call to `getOutputTimestamp` should wait until after we've established that the input timestamp is not `Left RebuildAlways`?

Reply:
I wouldn't think that would matter for this. `outputTimestamp` isn't forced unless it's compared. I suspect that it's actually getting forced as part of `prunePrebuiltModules`. Whereas before, it would process everything in order serially, and thus essentially short-circuit, maybe this is doing comparisons that it otherwise wouldn't have done?

Reply:
But yes, I can switch it around so it does it in a different order. It shouldn't check the output timestamp at all if it's RebuildAlways.

Reply:
I think what's happening is that in `findExistingExtern`, the new code no longer distinguishes the case where a module has a `RebuildNever` policy from the case where a module has no rebuild policy (i.e. it's just using timestamps as normal). In the former case, there's no need to compare the current module's output timestamp to its dependencies' output timestamps (because we already know we aren't going to bother rebuilding it). However, it looks like the `prunePrebuiltModules` in this PR treats both cases identically.

Reply:
Also I'm not sure about the name `prunePrebuiltModules`; the pruning isn't happening until the following line, right? I'd be more inclined to call that function something like `collectPrebuiltModules`.